### PR TITLE
set up local Ray.jl

### PR DESCRIPTION
**Pull request by kleinschmidt**
We have a repo for the user-facing Ray.jl already (https://github.com/beacon-biosignals/Ray.jl) but having separate repos for these two tightly coupled julia packages is inconvenient for initial rapid iteration.

Not _totally_ sure why we don't get QUITE as nice cache behavior as with the basic jll but it does go faster than the full build.

### CODE REVIEW

File: .github/workflows/Ray_CI.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+name: Ray.jl - CI
```
Comment (omus) [inline .github/workflows/Ray_CI.yml:1]: Can we rename the other CI workflow to include the package name just so these match?

File: .github/workflows/Ray_CI.yml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      # https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key
+      - uses: actions/cache/restore@v3
+        id: bazel-cache
+        with:
+          key: bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.hash-${{ hashFiles('deps/WORKSPACE.bazel.tpl', 'deps/BUILD.bazel') }}
+          path: ~/.cache/bazel
+          restore-keys: |
+            bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.
+            bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.
+            bazel-cache.${{ matrix.os }}.${{ matrix.arch }}.
```
Comment (omus) [inline .github/workflows/Ray_CI.yml:71]: The GHA cache action doesn't like to share caches between concurrently running jobs I would suggest doing:
```suggestion
          key: bazel-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.hash-${{ hashFiles('deps/WORKSPACE.bazel.tpl', 'deps/BUILD.bazel') }}
          path: ~/.cache/bazel
          restore-keys: |
            bazel-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.ray-${{ steps.ray-commit.outputs.head_sha }}.
            bazel-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.julia-${{ matrix.version }}.
            bazel-cache.ray-jl.${{ matrix.os }}.${{ matrix.arch }}.
```

File: Ray.jl/test/runtests.jl [hunk 3]
```diff
@@ -0,0 +1,9 @@
+include("set_up_tests.jl")
```
Comment (omus) [inline Ray.jl/test/runtests.jl:1]: Better name? Is there a precedent for this?
```suggestion
include("import_deps.jl")
```

File: .github/workflows/Ray_CI.yml (hunk 1 above)
Comment (kleinschmidt) [reply .github/workflows/Ray_CI.yml:1]: eh, given that we're gonna split this out I think this is better (to make that separation less painful)

File: .github/workflows/Ray_CI.yml (hunk 2 above)
Comment (kleinschmidt) [reply .github/workflows/Ray_CI.yml:71]: ah that's an interesting point.  I'd hoped to actually re-use the jll cache; but i guess if we don't need to do _full_ re-builds then it's probably okay.

File: Ray.jl/test/runtests.jl (hunk 3 above)
Comment (kleinschmidt) [reply Ray.jl/test/runtests.jl:1]: yeah, this is a beacon package templates thing to make running tests locally less painful.  but I'm not sure it's needed here (no juggling AWS permissiosn or anything of the sort)

File: Ray.jl/docs/fix_doctests.jl [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+# they are correct (and be easily revertable if they do something wrong).
+
+using Documenter, Ray
+
+DocMeta.setdocmeta!(Ray, :DocTestSetup, :(using Ray);
+                    recursive=true)
+
+if get(ENV, "CI", "false") == "true" || success(`git diff --quiet`)
+    if ismissing(get(ENV, "AWS_PROFILE", missing))
+        @warn """You may need to set `ENV["AWS_PROFILE"] = ray_core_worker_julia_jll-ci` in order to successfully run the doctests"""
```
Comment (glennmoy) [inline Ray.jl/docs/fix_doctests.jl:15]: do we need to make this role (or similar)?

File: Ray.jl/docs/make.jl [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+makedocs(; modules=[Ray],
+         sitename="Ray.jl",
+         authors="Beacon Biosignals",
+         pages=["API Documentation" => "index.md"],
+         strict=true)
+
+deploydocs(; repo="github.com/beacon-biosignals/ray_core_worker_julia_jll.jl.git", push_preview=true,
+           devbranch="main", dirname="Ray",
+           versions=["stable" => "Ray-v^", "Ray-v#.#",
+                     "dev" => "dev"])
```
Comment (glennmoy) [inline Ray.jl/docs/make.jl:13]:
```suggestion
                     "dev" => "main"])
```
I assume?

File: Ray.jl/docs/Project.toml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+[deps]
```
Comment (glennmoy) [inline Ray.jl/docs/Project.toml:1]: isn't there normally a Manifest for the docs as well?

**Review by glennmoy — APPROVED** [review]

File: .github/workflows/Ray_CI.yml [hunk 7]
```diff
@@ -0,0 +1,93 @@
+name: Ray.jl - CI
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    tags: ["*"]
+    paths: ["Ray.jl/**", ".github/workflows/Ray_CI.yml"]
```
Comment (omus) [inline .github/workflows/Ray_CI.yml:8]:
```suggestion
    paths:
      - "Ray.jl/**"
      - ".github/workflows/Ray_CI.yml"
```

File: .github/workflows/Ray_CI.yml [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+    paths: ["Ray.jl/**",".github/workflows/Ray_CI.yml"]
```
Comment (omus) [inline .github/workflows/Ray_CI.yml:11]:
```suggestion
    paths:
      - "Ray.jl/**"
      - ".github/workflows/Ray_CI.yml"
```

File: Ray.jl/docs/fix_doctests.jl (hunk 4 above)
Comment (kleinschmidt) [reply Ray.jl/docs/fix_doctests.jl:15]: I should just remove all teh docs stuff for now, we're not using it and this is from a template for a Lab sub-package which doesn't really map well onto this

File: Ray.jl/docs/make.jl (hunk 5 above)
Comment (kleinschmidt) [reply Ray.jl/docs/make.jl:13]: it's the sub-directory where dev docs are built (only have to specify this if we wanna deploy docs in this sub-package).  added a breadcrumb to change these settings when we move to own repo

File: Ray.jl/docs/Project.toml (hunk 6 above)
Comment (kleinschmidt) [reply Ray.jl/docs/Project.toml]: I just nuked all the docs stuff sicne it doesn't make sense trying to build them right now (all the caching etc. would have to be set up in teh docs build)

File: Ray.jl/test/runtests.jl (hunk 3 above)
Comment (kleinschmidt) [reply Ray.jl/test/runtests.jl:1]: I removed this in #19

**Pull request merged**